src/preamble.js

```diff
@@ -433,7 +433,15 @@ function UTF8ArrayToString(u8Array, idx) {
   while (u8Array[endPtr]) ++endPtr;
 
   if (endPtr - idx > 16 && u8Array.subarray && UTF8Decoder) {
-    return UTF8Decoder.decode(u8Array.subarray(idx, endPtr));
+    var buf;
+#ifdef USE_PTHREADS
+    // Need to copy the data to a non-shared buffer as TextDecoder isn't compatible with SharedArrayBuffer.
+    // Uint8Array.from would be more readable, but isn't universally available.
```

Comment:
Hmm, I think this is a Chrome bug? There was a doc somewhere about which APIs should be SAB compatible, and I think I did read TextDecoder there, and iirc I did test that to work on Firefox. If we had a `MINIMUM_CHROME_VERSION` #ifdef we've been talking about, that could be used here, although right now probably nothing to change on this.

Reply:
I was seeing a specific error message along the lines of "Array Buffer cannot be Shared", which is why I added this.  Which seems like a deliberate choice rather than a bug, but it may be something the Chrome devs only intended to be a temporary restriction.